articles/architecture-scenarios/application/mobile-api.md

```diff
@@ -4,35 +4,290 @@ title: Mobile + API
 extract: Mobile Application which talks to an API. The application will use OpenID Connect with the Authorization Code Grant using Proof Key for Code Exchange (PKCE) to authenticate users.
 description: Explains the architecture scenario with a mobile application client communicating with an API.
-beta: true
+toc: true
 ---
 
 # Mobile + API
 
+In this scenario we will build a Timesheet API for a fictitious company named ExampleCo. The API will allow management of timesheet entries for an employee or a contractor.
+
+We will also be building a Mobile Application which will be used to view and log timesheet entries in the centralized timesheet database using the API.
+
+
+::: panel TL;DR
+* Auth0 provides API Authentication and Authorizaion as a means to secure access to API endpoints (see [API Authentication and Authorization](#api-authentication-and-authorization))
+* For authorizing a Mobile app user and granting access to the API, Auth0 supports the Proof Key for Code Exchange (see [Proof Key for Code Exchange](#proof-key-for-code-exchange-pkce))
+* Both the Mobile app and the API must be configured in the Auth0 Dashboard (see [Auth0 Configuration](#auth0-configuration))
+* The API is secured by ensuring that a valid Access Token is passed in the HTTP Authorization header when calls are made to the API (see [Implement the API](#implement-the-api))
+* The Auth0.Android SDK can be used to authorize the user of the Mobile app and obtain a valid Access Token which can be used to call the API (see [Authorize the User](#authorize-the-user))
+* The Mobile app can retrieve the user's profile information by decoding the ID Token (see [Get the User Profile](#get-the-user-profile))
+* The Mobile app provides the Access Token in the HTTP Authorization header when making calls to the API (see [Call the API](#call-the-api))
+* The Mobile app user's Access Token can be renewed to ensure the user does not have to log in again during a session (see [Renew the Token](#renew-the-token))
+:::
+
+## The Premise
+
+ExampleCo is a consulting startup company. Currently they have approximately 100 employees and they also outsource several activities to external contractors. All employees and external contractors are required to fill in their timesheets every week. 
+
+The company has built a timesheets application, a scenario we covered in [Single Sign-On for Regular Web Apps](/architecture-scenarios/application/web-app-sso). The internal employees use this web app to fill in their timesheets but the company wants to replace it with a Mobile application. The app will be used to log timesheet entries and send the data to the centralized timesheet database using the API.
+
+### Goals & Requirements
+
+ExampleCo wants to build a flexible solution. There are potential multiple clients which should be able to log timesheet entries, as well as batch processes which may upload timesheet entries from other, external systems.
```

Comment:
Who are the clients that want to log timesheet entries? You mean employees/contractors?